framework/tlvf/src/src/CmduMessageTx.cpp

```diff
@@ -13,62 +13,39 @@
 
 using namespace ieee1905_1;
 
-CmduMessageTx::CmduMessageTx(uint8_t *buff, size_t buff_len) : CmduMessage()
-{
-    m_buff     = buff;
-    m_buff_len = buff_len;
-}
+CmduMessageTx::CmduMessageTx(uint8_t *buff, size_t buff_len) : CmduMessage(buff, buff_len) {}
 
 CmduMessageTx::~CmduMessageTx() {}
 
 std::shared_ptr<cCmduHeader> CmduMessageTx::create(uint16_t id, eMessageType message_type)
 {
-    if (!m_buff || !m_buff_len)
+    // parse - false since this is for TX
+    // swap - false since on TX we do swap in Finalize(swap_needed)
+    msg.reset(false, false);
+    auto cmduhdr = addClass<cCmduHeader>();
+    if (!cmduhdr)
         return nullptr;
 
-    m_parse = false;
-    memset(m_buff, 0, m_buff_len);
-    reset();
-    m_cmdu_header = std::make_shared<cCmduHeader>(m_buff, m_buff_len);
-    if (!m_cmdu_header || m_cmdu_header->isInitialized() == false) {
-        return nullptr;
-    }
-
-    m_cmdu_header->message_type() = message_type;
-    m_cmdu_header->message_id()   = id;
-    return m_cmdu_header;
+    cmduhdr->message_type() = message_type;
+    cmduhdr->message_id()   = id;
+    return cmduhdr;
 }
 
 std::shared_ptr<cCmduHeader> CmduMessageTx::load()
 {
-    if (!m_buff || !m_buff_len)
-        return nullptr;
-
-    m_parse = true;
-    reset();
-    m_cmdu_header = std::make_shared<cCmduHeader>(m_buff, m_buff_len, m_parse);
-    if (!m_cmdu_header || m_cmdu_header->isInitialized() == false) {
-        return nullptr;
-    }
-
-    return m_cmdu_header;
+    msg.reset(true, false);
+    return addClass<cCmduHeader>();
```

Comment:
No need for the copy constructor. The latest works - you are looking at a broken commit.
Since this change is so big, I had to include some commits which break things so it would be reviewable. I agreed with Arnout to keep the broken commits so in the future it would be easy to review this change for someone who is not familiar with TLVF.